integration-tests/src/tests/client/resharding.rs

```diff
@@ -850,25 +867,30 @@ fn test_shard_layout_upgrade_simple_impl(resharding_type: ReshardingType, rng_se
 
 #[test]
 fn test_shard_layout_upgrade_simple_v1() {
-    test_shard_layout_upgrade_simple_impl(ReshardingType::V1, 42);
+    test_shard_layout_upgrade_simple_impl(ReshardingType::V1, 42, false);
```

Comment:
IIUC this test covers the `ForReshardingOnly` code branch, but is identical to `test_shard_layout_upgrade_simple_v1_with_snapshot_enabled` in every other aspect.

Reply:
Yes, in the integration tests, I've kept the snapshotting config to `ForReshardingOnly` by default. The `test_shard_layout_upgrade_simple_v1_with_snapshot_enabled` test covers the code path for snapshotting config `EveryEpoch`